### PR TITLE
Switch with_dbpath from CString to String

### PR DESCRIPTION
**Pull request by ilpianista**
I find it very unusual for a third-party client to use CString.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -111,20 +111,18 @@ pub struct Alpm {
...
+  pub fn with_dbpath(dbpath: String) -> Result<Alpm, std::io::Error> {
     let lib = try!( so::Library::new("/usr/lib/libalpm.so") );
 
-
     let root = try!( CString::new("/") );
     let mut error_no = Box::new(0);
     let handle = unsafe {
       let init: Symbol<unsafe extern fn(*const c_char, *const c_char, *mut usize) -> *const usize> = try!(lib.get(b"alpm_initialize\0"));
-      init(root.as_ptr(), dbpath.as_ptr(), error_no.as_mut())
+      init(root.as_ptr(), CString::new(dbpath).unwrap().as_ptr(), error_no.as_mut())
```
Comment (dns2utf8) [inline src/lib.rs]: Does this leak the CString object?

**Review by dns2utf8 — CHANGES_REQUESTED** [review]
You are right. Good point

File: src/lib.rs (hunk 1 above)
Comment (ilpianista) [reply src/lib.rs]: Sincerely I don't know. Better now?

Comment (dns2utf8) [reply src/lib.rs]: This is safe with the scope rules I know. Thank you!

**Review by dns2utf8 — APPROVED** [review]

**Pull request merged**